godot-macros/src/derive/derive_from_variant.rs

```diff
@@ -24,8 +24,8 @@ pub fn derive_from_godot(decl: Declaration) -> ParseResult<TokenStream> {
     } = decl_get_info(&decl);
 
     let mut body = quote! {
-        let root = variant.try_to::<::godot::builtin::Dictionary>().ok()?;
-        let root = root.get(#name_string)?;
+        let root = variant.try_to::<::godot::builtin::Dictionary>()?;
+        let root = root.get(#name_string).ok_or(ConvertError::custom().with_cause(concat!("missing expected value ", #name_string)).with_value(root.clone()))?;
```

Comment:
Instead of `concat!` in the generated code, the string could be concatenated outside of `quote!`.

Also, splitting `root.get(#name_string)` into a separate variable would allow to avoid the `root.clone()`, no? Since `get` returns a value and doesn't borrow `root` any longer.

Reply:
sure though i was planning on rewriting this whole derive macro soon anyway

Reply:
Ah ok, in that case no need to bother 🙂